build/cmake/CMakeLists.txt

```diff
@@ -2,11 +2,18 @@ cmake_minimum_required(VERSION 3.20)
 
 # This is umbrella CMakeLists.txt for all subrepos of the monorepo.
 
-if(NOT DEFINED ENV{VCPKG_ROOT})
-    message(FATAL_ERROR "VCPKG_ROOT is not set")
+# We take VCPKG_ROOT from the environment, but we have to store it in CMake cache, so that if IDE later
+# needs to rerun configuration phase, it will still work even if the environment variable is not set anymore.
+if(NOT VCPKG_ROOT)
+    if(NOT DEFINED ENV{VCPKG_ROOT})
+        message(FATAL_ERROR "VCPKG_ROOT is not set")
+    endif()
+    set(VCPKG_ROOT "$ENV{VCPKG_ROOT}" CACHE INTERNAL "" FORCE)
+else()
+    set(ENV{VCPKG_ROOT} "${VCPKG_ROOT}")
```

Comment:
I'm not sure I understand how it works here. You set the environment variable? Is it necessary for some use case?